postpic/particles/particles.py

```diff
@@ -1279,3 +1279,38 @@ def collect(self, *scalarfs):
                 particlelist[i].append(scalars[:, k])
         ret = [np.asarray(p).T for p in particlelist]
         return ret
+
+    def exportToHdf5(self, filename, *scalarfs):
+        '''
+        Exports the collected data to an hdf5-file. If the file with the same name
+        already exists, it will be deleted!
+        Parameters:
+        -----------
+        filename:   whole name or path of the file which the data will be exported to.
+        *scalarfs:  the scalarfunction(s) defining the particle property.
+                Id and weight will always be stored, so they do not have to be
+                specified here.
+
+        Returns:
+        -----------
+        The same as collect(): The list of particles with the defined properties.
+        The hdf5 file will have the structure:
+        - dset: Ids
+        - particles/Id/track: track data (ID, weight, scalarfs)
+        - particles/Id/weight: weights
+        '''
+        scalarfs = ("id", "weight") + scalarfs
+        part_data = self.collect(*scalarfs)
+
+        import h5py
+
+        f = h5py.File(filename, "w")
```

Comment:
Instead of opening and closing the file manually, can you please use the with statement to ensure the file is closed even when errors occur during writing?
```
with h5py.File(filename, "w") as f:
.....
```